libs/langchain/langchain/memory/chat_message_histories/streamlit.py

```diff
@@ -0,0 +1,35 @@
+from typing import List
+
+from langchain.schema import (
+    BaseChatMessageHistory,
+)
+from langchain.schema.messages import BaseMessage
+
+
+class StreamlitChatMessageHistory(BaseChatMessageHistory):
+    """
+    Chat message history that stores messages in Streamlit session state.
+
+    Args:
+        key: The key to use in Streamlit session state for storing messages.
+    """
+
+    def __init__(self, key: str = "messages"):
+        import streamlit as st
+
+        if key not in st.session_state:
+            st.session_state[key] = []
```

Comment:
nit: @sfc-gh-jcarroll maybe a good idea is if a key already exists in `session_state` override it with an empty array, or at least check that it is a Python list. 

I think potentially we could have a situation when under the `messages` key we have e.g. dict from streamlit app, and then it could fail on `append` in the `add_message` method.

Reply:
I think it adds more complexity and it's OK for the developer to manage it. It will get more tricky to assess across re-runs what is happening.

Two simple options:
* Check on __init__() only if there's an existing key which is not a list, and throw a warning in that case
* Use a less obvious default key than `"messages"` which is less likely to have a conflict

But honestly I think the current version is fine, easy enough to debug unless / until we get developer feedback about it.

@LukasMasuch thoughts too?

Also @sfc-gh-kjavadyan particularly interested in what you think on the test implementation and maintenance. Thanks!

Reply:
No strong opinion from my side. I would probably go for a more specific default key- maybe `langchain_messages` - which makes it a lot less likely for a problem to occur and it is also more obvious what it contains in case a user inspects the session state.

Reply:
Switched key to `langchain_messages` as suggested, to reduce the risk of a confusing key conflict.